app/controllers/ish_manager/venues_controller.rb

```diff
@@ -14,34 +14,45 @@ def index
 
   def new
     @venue = Venue.new
+    @venue.city_id = params[:city_id] if params[:city_id]
+    @venue.to_city = 'yes' if params[:city_id]
+
     authorize! :new, @venue
   end
 
   def create
+    redirect_path = venues_path()
     @venue = Venue.new params[:venue].permit!
     authorize! :create, @venue
+    if params[:venue][:to_city] == 'yes'
+      redirect_path = city_path( params[:venue][:city_id] )
+    end
     if @venue.save
-      redirect_to :action => :index
+      redirect_to redirect_path, :notice => 'Venue successfully created.'
     else
-      flash[:alert] = @venue.errors.messages
+      #flash[:alert] = @venue.errors.messages
       render :action => :new
     end
   end
 
   def edit
     @venue = Venue.find params[:id]
+    @venue.to_city = params[:from_city]
     authorize! :edit, @venue
   end
 
   def update
+
+    redirect_path = venues_path()
+    redirect_path = city_path( params[:venue][:city_id] ) if params[:venue][:to_city] == 'yes'
     params[:venue][:tag_ids].delete ''
     @resource = @venue = Venue.find params[:id]
     authorize! :update, @venue
     update_profile_pic
     flag = @venue.update_attributes params[:venue].permit!
     if flag
       flash[:notice] = 'updated venue'
-      redirect_to :action => :index
```

Comment:
let's change it to, always redirects back to the city (never to the index of venues)